assert.js

```diff
@@ -596,6 +596,52 @@ assert.ifError = function ifError(err) {
   }
 };
 
+// Currently in sync with Node.js lib/assert.js
+// https://github.com/nodejs/node/commit/2a871df3dfb8ea663ef5e1f8f62701ec51384ecb
+function internalMatch(string, regexp, message, fn, fnName) {
+  if (!isRegExp(regexp)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'regexp', 'RegExp', regexp
+    );
+  }
+  const match = fnName === 'match';
+  const test = require('call-bind/callBound')('RegExp.prototype.test');
```

Comment:
i meant that as quasi pseudocode :-) the require and this line should both be done at module level (`var callBound = require('call-bind/callBound');`) not inside this runtime function.

Reply:
@ljharb :facepalm: of course! I moved it to the module level and named it `RegExpPrototypeTest` instead (like we name primordials in Node.js) and used it wherever the `test` functionality was being used. Not very sure why we should use `var` though, given that it is very sparingly used in this module. Could u please explain?

Reply:
This entire package is a browser shim, for older engines; if other things are using const than obv you can use it too, but probably none of it should be.

Reply:
Cool, I'll do that in a follow-up PR :+1: